programs/adapter-jupiter/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::{
+    hash::hash,
+    instruction::{AccountMeta, Instruction},
+    program::invoke,
+    pubkey::Pubkey,
+};
+use anchor_spl::token::TokenAccount;
+
+declare_id!("ADPT8iF4A7BSUWQ8AsVwmcod2suFzA4bpYpJj7kUWK3E");
+
+const SWAP_CONFIG_SIZE: usize = 30;
+
+#[program]
+pub mod adapter_jupiter {
+    use super::*;
+
+    const PLATFORM_FEE: u8 = 0;
+
+    pub fn swap(ctx: Context<Action>, input: Vec<u8>) -> Result<()> {
+        let discriminator: [u8; 8] = sighash("global", "route");
+
+        // Use remaining accounts
+        let mut dest_token_account_and_balance =
+            load_token_account_and_balance(ctx.remaining_accounts, 2);
+
+        // Get Input
+        let mut input_bytes: &[u8] = &input[..];
+        let input_struct = SwapInputWrapper::deserialize(&mut input_bytes)?;
+        msg!("Input: {:?}", input_struct);
+
+        let mut last_index: usize = SWAP_CONFIG_SIZE - 1;
+        let mut start_index: usize = 1;
+        if input_struct.swap_config[0] > 0 && input_struct.swap_config[0] < SWAP_CONFIG_SIZE as u8 {
+            last_index = input_struct.swap_config[0] as usize;
+        } else if input_struct.swap_config[0] == 0 {
+            start_index = 0;
+        };
+
+        let swap_accounts = load_remaining_accounts(ctx.remaining_accounts, None);
+
+        let mut swap_data = vec![];
+        swap_data.append(&mut discriminator.try_to_vec()?);
+        swap_data.extend(
+            &mut input_struct.swap_config[start_index..=last_index]
```

Comment:
Just wonder: what is start_index and last_index for

Reply:
For extracting swap config
Jupiter swap's ix data contain 5 props in this order:
a). SwapLeg: (uncertain length) 
b) inAmount: `u64` 
c) outAmount: `u64` 
d) slippage: `u16` 
e) platform fee: `u8` 

since we can only set `swap_config` a fix `u8` array size in `SwapInwrapper`,
we need to extract the subarray from this attribute to fit the payload format.

Reply:
Noted